### ISSUE TITLE
Fails to retrieve digest from Red Hat registry

### ISSUE THREAD

**Issue by lomori** [genuinetools/reg#161]
See commands below (no credentials needed for that image):

$ reg digest registry.access.redhat.com/rhscl/mysql-57-rhel7
invalid checksum digest format

$ docker pull registry.access.redhat.com/rhscl/mysql-57-rhel7
Using default tag: latest
latest: Pulling from rhscl/mysql-57-rhel7
Digest: sha256:6652ef182f61bf1efa25ea101346d6bf72685ac897902a837cf73b200e3a5b2d
Status: Image is up to date for registry.access.redhat.com/rhscl/mysql-57-rhel7:latest

version tested: v0.15.8
environment: MacOS High Sierra 10.13.6

**Comment by rikkuness** [comment]
It seems that the RedHat registry does not add the header `Docker-Content-Digest` on the response when asking for a manifest which is what reg is checking.

https://github.com/genuinetools/reg/blob/master/registry/digest.go#L38

<img width="605" alt="screen shot 2018-11-08 at 13 11 38" src="https://user-images.githubusercontent.com/3525311/48200802-26a86f00-e358-11e8-8b02-a60efdd12376.png">

**Comment by lomori** [comment]
Yes, I've noticed that also. I wonder how the docker cli is doing. It does display it upon pulling so does it need to pull the image to check the digest or is it figuring that out some other way?
Regards,Luiz
   On Thursday, November 8, 2018, 8:09:21 AM EST, Darrian <notifications@github.com> wrote: 
 
 
It seems that the RedHat registry does not add the header Docker-Content-Digest on the response when asking for a manifest which is what reg is checking.

https://github.com/genuinetools/reg/blob/master/registry/digest.go#L38

**Comment by rikkuness** [comment]
The digest is also in the manifest.

I had a go at changing the `Digest` method to use it from the manifest, which is quite elegant but I'm not sure how that accounts for version 1 manifests.

```go
// Digest returns the digest for an image.
func (r *Registry) Digest(image Image) (digest.Digest, error) {
	if len(image.Digest) > 1 {
		// return early if we already have an image digest.
		return image.Digest, nil
	}

	mf, err := r.ManifestV2(image.Path, image.Reference())
	if err != nil {
		return "", err
	}

	return mf.Config.Digest, nil
}
```

**Comment by ricardobranco777** [comment]
> ```go
> .Config.Digest
> ```

This is not the Digest but the Docker ID.

**Comment by ricardobranco777** [comment]
> The digest is also in the manifest.
> 
> I had a go at changing the `Digest` method to use it from the manifest, which is quite elegant but I'm not sure how that accounts for version 1 manifests.
> 
> ```go
> // Digest returns the digest for an image.
> func (r *Registry) Digest(image Image) (digest.Digest, error) {
> 	if len(image.Digest) > 1 {
> 		// return early if we already have an image digest.
> 		return image.Digest, nil
> 	}
> 
> 	mf, err := r.ManifestV2(image.Path, image.Reference())
> 	if err != nil {
> 		return "", err
> 	}
> 
> 	return mf.Config.Digest, nil
> }
> ```

This is not the Digest but the Docker ID.
